Python/compile.c

```diff
@@ -5421,6 +5452,601 @@ compiler_slice(struct compiler *c, expr_ty s)
     return 1;
 }
 
+
+// PEP 634: Structural Pattern Matching
+
+// To keep things simple, all compiler_pattern_* routines follow the convention
+// of preserving TOS (the subject for the given pattern) and pushing either True
+// (match) or False (no match) on top of it. We do this even for irrefutable
+// patterns; the idea is that it's much easier to smooth out any redundant
+// pushing, popping, and jumping in the peephole optimizer than to detect or
+// predict it here.
+
+
+#define WILDCARD_CHECK(N) \
+    ((N)->kind == Name_kind && \
+    _PyUnicode_EqualToASCIIString((N)->v.Name.id, "_"))
+
+
+static int
+pattern_helper_load_attr(struct compiler *c, expr_ty p, pattern_context *pc)
```

Comment:
How does this differ from `compiler_visit_expr1(c, p)` when passed an attribute expression?
I suspect that you could replace all uses of this with `compiler_visit_expr1(c, p)`.